packages/EasyDoctrine/src/Dispatchers/DeferredEntityEventDispatcher.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyDoctrine\Dispatchers;
+
+use EonX\EasyDoctrine\Events\EntityCreatedEvent;
+use EonX\EasyDoctrine\Events\EntityUpdatedEvent;
+use LogicException;
+use Psr\EventDispatcher\EventDispatcherInterface;
+
+final class DeferredEntityEventDispatcher implements DeferredEntityEventDispatcherInterface
+{
+    /**
+     * @var bool
+     */
+    private $enabled;
+
+    /**
+     * @var array<int, array<string, array<string, array{mixed, mixed}>>>
+     */
+    private $entityChangeSets = [];
+
+    /**
+     * @var array<string, object>
+     */
+    private $entityInsertions = [];
+
+    /**
+     * @var array<string, object>
+     */
+    private $entityUpdates = [];
+
+    /**
+     * @var \Psr\EventDispatcher\EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    public function __construct(EventDispatcherInterface $eventDispatcher)
+    {
+        $this->enabled = true;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function clear(?int $transactionNestingLevel = null): void
```

Comment:
I really like this function, and the fact that we're calling it after dispatching.

But if anything goes wrong between `onFlush` and `postFlush` then there is a risk that this function won't be called, and it won't be an issue for normal requests as PHP process will be killed and its memory with it.

However, I can see this becoming an issue in the queue worker context as the PHP process won't be killed every time, and we might have entities from a previously failed message still waiting to be dispatched.
They might be dispatched when they're not supposed to, or the memory might build up until the worker "explodes" :smile: 

I think that the `EntityManagerDecorator` should override the `clear()` method and call this `clear()` method on the `DeferredEntityEventDispatcher` as we already have event listeners calling the entity manager clear method each time a new message is received.